src/Graph.php

```diff
@@ -1,184 +1,172 @@
-	 *
-	 * @param string $requestType The HTTP method to use, e.g. "GET" or "POST"
-	 * @param string $endpoint    The Graph endpoint to call
-	 *
-	 * @return GraphCollectionRequest The request object, which can be
-	 *                                used to make queries against Graph
-	 * @throws Exception\GraphException
-	 */
-    public function createCollectionRequest($requestType, $endpoint)
-    {
-        return new GraphCollectionRequest(
-            $requestType,
-            $endpoint,
-            $this->_accessToken,
-            $this->_baseUrl,
-            $this->_apiVersion,
-            $this->_proxyPort,
-            $this->_proxyVerifySSL
-        );
-    }
-}
+<?php
+/**
+* Copyright (c) Microsoft Corporation.  All Rights Reserved.
+* Licensed under the MIT License.  See License in the project root
+* for license information.
+*
+* Graph File
+* PHP version 7
+*
+* @category  Library
+* @package   Microsoft.Graph
+* @copyright 2016 Microsoft Corporation
+* @license   https://opensource.org/licenses/MIT MIT License
+* @version   GIT: 0.1.0
+* @link      https://graph.microsoft.io/
+*/
+
+namespace Microsoft\Graph;
+
+use Microsoft\Graph\Core\GraphConstants;
+use Microsoft\Graph\Core\NationalCloud;
+use Microsoft\Graph\Exception\ClientInitialisationException;
+use Microsoft\Graph\Http\GraphCollectionRequest;
+use Microsoft\Graph\Http\GraphRequest;
+use Microsoft\Graph\Http\HttpClientFactory;
+use Microsoft\Graph\Http\HttpClientInterface;
+
+/**
+ * Class Graph
+ *
+ * @category Library
+ * @package  Microsoft.Graph
+ * @license  https://opensource.org/licenses/MIT MIT License
+ * @link     https://graph.microsoft.io/
+ */
+class Graph
+{
+    /**
+    * The access_token provided after authenticating
+    * with Microsoft Graph (required)
+    *
+    * @var string
+    */
+    private $_accessToken;
+
+    /**
+    * The api version to use ("v1.0", "beta")
+    * Default is "v1.0"
+    *
+    * @var string
+    */
+    private $_apiVersion = GraphConstants::API_VERSION;
+
+    /**
+     * Host to use as the base URL and for authentication
+     * @var string
+     */
+    private $_nationalCloud = NationalCloud::GLOBAL;
+    
+    /**
+     * HttpClient to use for requests
+     * @var HttpClientInterface
+     */
+    private $_httpClient;
+
+
+    // TODO: Set middleware
+
+    /**
+     * Graph constructor.
+     *
+     * Creates a Graph client object used to make requests to the Graph API
+     * @param string $apiVersion
+     * @param string $nationalCloud
+     * @throws \InvalidArgumentException|Exception\ClientInitialisationException
+     */
+    public function __construct(string $apiVersion = GraphConstants::API_VERSION,
+                                string $nationalCloud = NationalCloud::GLOBAL)
+    {
+        if (!$apiVersion) {
+            throw new \InvalidArgumentException("Api version string cannot be empty");
+        }
+        if (!in_array($nationalCloud, NationalCloud::getValues())) {

```

Comment:
No action, just a comment. 

The one issue we could (in another library) see here is if someone were to use the client in a pre-production environment, they may need to set a different endpoint than the ones in the national cloud. I don't expect to see pre-production testing with this library.

Reply:
Nice to know, thanks.